contract_show_recurring_invoice/views/contract_view.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+
+        <record id="act_recurring_invoices" model="ir.actions.act_window">
+            <field name="context">{'search_default_contract_id':
+                [active_id],
+                'default_contract_id': active_id}
+            </field>
+            <field name="name">Invoices</field>
+            <field name="res_model">account.invoice</field>
+            <field name="domain">[('contract_id','=',active_id)]</field>
```

Comment:
I think the filter is defined twice: the domain "hard" filter and the context "search_default" soft filter.
Maybe you can drop the domain filter.